online-shop/src/app/services/user.service.ts

```diff
@@ -0,0 +1,35 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from "@angular/common/http";
+import {CredentialsModel} from "../types/credentials.model";
+import {Observable, tap} from "rxjs";
+import {Role, User} from "../types/user.model";
+import {url} from "../utils";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  currentUser: User | undefined;
+
+  redirectUrl: string | null = null;
+
+  constructor(private http: HttpClient) {
+  }
+
+  login(credentials: CredentialsModel): Observable<User> {
+    return this.http.post<User>(`${url}/login`, credentials).pipe(tap((user) => {
+      {
+        this.currentUser = user
+        localStorage.setItem("username", this.currentUser.username)
+      }
+    }))
+  }
+
+  isLoggedIn() {
+    return this.currentUser != undefined
```

Comment:
**suggestion:** always use the strict equality comparison (`===` and `!==`). https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness